e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js

```diff
@@ -88,4 +77,465 @@ describe('Profile popover', () => {
             cy.get('#add-user-to-channel-modal__add-button').should('be.disabled');
         });
     });
+
+    it('MM-T3 Add user — Public ON / Private OFF', () => {
+        // # Do the system console setup
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+        cy.findByTestId('all_users-private_channel-checkbox').scrollIntoView().should('be.visible').click();
+
+        // * Verify that all the sub-checkboxes are disabled
+        verifyPermissionSubSections('all_users', 'private', false);
+
+        cy.findByTestId('saveSetting').should('be.visible').click();
+
+        cy.apiLogout();
+
+        // # Login as the second user now
+        cy.apiLogin(otherUser);
+
+        // # Create a private channel
+        createPrivateChannel(testTeam.id, otherUser).then((channel) => {
+            privateChannel = channel;
+        });
+
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        clickAddToChannel(testUser);
+
+        cy.get('div[aria-labelledby="addChannelModalLabel"]').within(() => {
+            // # Type "private" and press enter.
+            cy.get('input').should('be.visible').type('private').wait(TIMEOUTS.HALF_SEC).type('{enter}');
+
+            // # Verify that button is disabled
+            cy.get('#add-user-to-channel-modal__add-button').should('be.disabled');
+        });
+    });
+
+    it('MM-T4 Add user — Public OFF / Private ON', () => {
+        // # Do the system console setup
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+        cy.findByTestId('all_users-public_channel-checkbox').scrollIntoView().should('be.visible').click();
```

Comment:
Looks like this one requires another click to fully uncheck the "Manage Public Channels". Initial state has "Convert Channels" uncheck on me and so first click makes the entire "Manage Public Channels" checked.

Reply:
Ah looks like that was it. I had an outdated copy of the webapp. Fixing.